### ISSUE TITLE
WithMuzzleOverlay is broken for multiple armaments per actor

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#10763]
In my case an alpha palette that works perfectly fine on explosions doesn't work at all on muzzle flashes even when using the same shp. All sequences are set up the same way (BlendMode: Additive, no other custom settings). I did check my yaml for typos or other stupidities at least 5 times and tried everything I could think of.

Muzzle flashes get brighter, but transparency stays zero, and the palette's Alpha value is completely ignored.

**Comment by Apollo-X** [comment]
Does this additive and alpha value combination ever work? What I've read whole principle of additive blending is not to use alpha blending principles.

**Comment by reaperrr** [comment]
It does work on explosions, for example.
But I tested muzzles with just alpha as well, same result. it seems blending/transparency in general doesn't work on muzzles. Additive makes the colors brighter, but it doesn't blend them with the background.

**Comment by reaperrr** [comment]
@pchote
Actually, alpha blending works fine. The reason why it looked glitchy is that `WithMuzzleOverlay` seems to be really broken. It uses the palettes of all defined `Armament`s at once.
And that means if you add an `Armament` that does not define a custom palette, even if that `Armament` has not muzzle effect, its default `MuzzlePalette` (`effect`) will be used anyhow. Generally speaking it also seems to draw the last defined `Armament`'s muzzle on top, but if that last `Armament`'s `MuzzlePalette` uses alpha transparency, you can see the muzzle drawn in default `effect` palette below it.

Basically, it's probably been broken ever since the trait isn't defined for every individual Armament anymore, there just weren't any mod setups that exposed it, until now.

**Comment by reaperrr** [comment]
In other words, the reason why blending didn't seem to work was that I had two armaments defined on the actor and the muzzle was displayed twice on the same spot, which obviously made it even brighter due to blending. And no, the armaments were not using the same offsets.

**Comment by DoGyAUT** [comment]
Any chance this might get looked into?

I have quite a few units which use multiple weapons with different muzzle pals. I helped myself to convert them to the same pal, but it's still not an optimal solution, as I can't do that with each animation.

**Comment by pchote** [comment]
Can somebody please provide a test case and screenshots of how to spot the problem?  I see no diference in #13080 before and after removing the "TODO: remove once WithMuzzleOverlay has been fixed"s
